### PR TITLE
[raster][UX] Raster identify: respect locale settings

### LINKED ISSUE #57765: Identify Results for raster data value does not use correct decimal separator
### What is the bug or the crash?

When using the Identify tool on a raster layer the value is apparently always shown using decimal point even on systems configured to use decimal comma.
image

### Steps to reproduce the issue

1) Load a raster layer with floating point values. 
2) Click a cell with the Identify tool
3) Check the Identify Results panel

### Versions

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><style type="text/css">
p, li { white-space: pre-wrap; }
</style></head><body>

QGIS version | 3.36.3-Maidenhead | QGIS code revision | 2df9655469b
-- | -- | -- | --
Qt version | 5.15.2
Python version | 3.9.5
GDAL/OGR version | 3.3.2
PROJ version | 8.1.1
EPSG Registry database version | v10.028 (2021-07-07)
GEOS version | 3.9.1-CAPI-1.14.2
SQLite version | 3.35.2
PDAL version | 2.3.0
PostgreSQL client version | unknown
SpatiaLite version | 5.0.1
QWT version | 6.1.6
QScintilla2 version | 2.11.5
OS version | macOS 14.5
  |   |   |  
Active Python plugins
FreehandRasterGeoreferencer | 0.8.3
firstaid | 3.1.1
debug_vs | 0.8
geosearch_dk | 2.0.4
OneDoor | 0.2.32
zoom_level | 0.3.1
profiletool | 4.2.6
trackable_project_files | v1.2.2
DataPlotly | 4.1.0
IPyConsole | version 2.0
dmpcatalogue | 1.1.1
linking_relation_editor | v1.3.0
QgisModelBaker | v7.8.4
qgis_stac | 1.1.2
AddressToolsDK | 0.1
Dataforsyningen | 0.4
OSMDownloader | 1.0.3
pgtopoeditor | 0.3.1
QlrBrowser | 3.0.0
plugin_reloader | 0.9.4
valuetool | 3.0.19
processing | 2.12.99
grassprovider | 2.12.99
db_manager | 0.1.20

</body></html>QGIS version
3.36.3-Maidenhead
QGIS code revision
[2df9655469b](https://github.com/qgis/QGIS/commit/2df9655469b)
Qt version
5.15.2
Python version
3.9.5
GDAL/OGR version
3.3.2
PROJ version
8.1.1
EPSG Registry database version
v10.028 (2021-07-07)
GEOS version
3.9.1-CAPI-1.14.2
SQLite version
3.35.2
PDAL version
2.3.0
PostgreSQL client version
unknown
SpatiaLite version
5.0.1
QWT version
6.1.6
QScintilla2 version
2.11.5
OS version
macOS 14.5

Active Python plugins
FreehandRasterGeoreferencer
0.8.3
firstaid
3.1.1
debug_vs
0.8
geosearch_dk
2.0.4
OneDoor
0.2.32
zoom_level
0.3.1
profiletool
4.2.6
trackable_project_files
v1.2.2
DataPlotly
4.1.0
IPyConsole
version 2.0
dmpcatalogue
1.1.1
linking_relation_editor
v1.3.0
QgisModelBaker
v7.8.4
qgis_stac
1.1.2
AddressToolsDK
0.1
Dataforsyningen
0.4
OSMDownloader
1.0.3
pgtopoeditor
0.3.1
QlrBrowser
3.0.0
plugin_reloader
0.9.4
valuetool
3.0.19
processing
2.12.99
grassprovider
2.12.99
db_manager
0.1.20

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by elpaso**
Fix #57765

### CODE REVIEW

File: src/gui/qgsmaptoolidentify.cpp [hunk 1]
```diff
@@ -1176,11 +1176,17 @@ bool QgsMapToolIdentify::identifyRasterLayer( QList<IdentifyResult> *results, Qg
           // the return value should be interpreted as QMetaType::Type"
           if ( static_cast<QMetaType::Type>( value.userType() ) == QMetaType::Float )
           {
-            valueString = QgsRasterBlock::printValue( value.toFloat() );
+            const float fValue { value.toFloat() };
+            valueString = QgsRasterBlock::printValue( fValue );
```
Comment (nyalldawson) [inline src/gui/qgsmaptoolidentify.cpp:1180]: QgsRasterBlock::printValue isn't exposed to Python, and this is the ONLY use of it. I'd just move the locale handling into that function directly...

Comment (elpaso) [reply src/gui/qgsmaptoolidentify.cpp:1180]: Actually, I checked and there are multiple uses of that function, not all of them expect a localized string.

```text
grep -ril 'QgsRasterBlock::printValue' src|egrep -v 'sorti|prepare'
src/providers/grass/qgsgrassrasterprovider.cpp
src/core/raster/qgsrasterblock.cpp
src/core/raster/qgscontrastenhancement.cpp
src/core/raster/qgsrastertransparency.cpp
src/core/raster/qgssinglebandpseudocolorrenderer.cpp
src/core/raster/qgsrasterlayer.cpp
src/core/raster/qgscolorrampshader.cpp
src/gui/raster/qgsrastertransparencywidget.cpp
src/gui/raster/qgsrasterlayerproperties.cpp
src/gui/qgsrasterlayersaveasdialog.cpp
src/gui/qgsmaptoolidentify.cpp
```

Comment (nyalldawson) [reply src/gui/qgsmaptoolidentify.cpp:1180]: Argh sorry, I was looking at the float version only. Indeed the double version IS exposed to Python.

Could we add an optional `bool respectLocale = false` argument to printValue though?

Comment (elpaso) [reply src/gui/qgsmaptoolidentify.cpp]: Done, test added.

File: src/core/raster/qgsrasterblock.cpp [hunk 2]
```diff
@@ -699,8 +701,14 @@ QString QgsRasterBlock::printValue( double value )
...
   {
     s.setNum( value, 'g', i );
-    if ( qgsDoubleNear( s.toDouble(), value ) )
+    const double doubleValue { s.toDouble( ) };
+    if ( qgsDoubleNear( doubleValue, value ) )
     {
+      if ( localized )
+      {
+        const int precision { static_cast<int>( s.count( QRegularExpression( QStringLiteral( "\\d" ) ) ) ) };
+        return QLocale().toString( doubleValue, 'g', precision );
```
[the remark this answers is not in the document]
Comment (elpaso) [reply src/core/raster/qgsrasterblock.cpp]: done

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**